presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -265,7 +279,17 @@ private void rehash()
         if (newCapacityLong > Integer.MAX_VALUE) {
             throw new PrestoException(GENERIC_INSUFFICIENT_RESOURCES, "Size of hash table cannot exceed 1 billion entries");
         }
-        int newCapacity = (int) newCapacityLong;
+        int newCapacity = toIntExact(newCapacityLong);
+
+        // An estimation of how much extra memory needed before allocating the hash table.
```

Comment:
An **estimate** of how much extra memory **is** needed before we can go ahead and **expand** the hash table.

(3 changes)

> An estimate is an approximate calculation or evaluation, and an estimation is the process of approximately calculating or evaluating